python/mpt/src/mpt/state_diff.py

```diff
@@ -0,0 +1,242 @@
+import logging
+from typing import Dict, Optional
+
+from ethereum.cancun.fork_types import EMPTY_ACCOUNT, Account, Address
+from ethereum.cancun.state import State, get_account, get_storage
+from ethereum_types.bytes import Bytes32
+from ethereum_types.numeric import U256
+
+logger = logging.getLogger()
+
+
+class AccountDiff:
+    """
+    A class that contains the differences for a single account, including its storage changes.
+    If account is None, it represents a deletion.
+    """
+
+    account: Optional[Account]
+    storage_updates: Dict[Bytes32, U256]
+
+    def __init__(self, account: Optional[Account] = None):
+        self.account: Optional[Account] = account
+        self.storage_updates: Dict[Bytes32, U256] = {}
+
+
+class StateDiff:
+    """
+    A class that contains the differences between two states.
+    """
+
+    account_diffs: Dict[Address, AccountDiff]
+
+    def __init__(self):
+        self.account_diffs: Dict[Address, AccountDiff] = {}
+
+    @staticmethod
+    def from_pre_post(pre_state: State, post_state: State) -> "StateDiff":
+        """
+        Compute a state diff between two states: a pre_state and a post_state.
+
+        Parameters
+        ----------
+        pre_state : State
+            The state before executing a block
+        post_state : State
+            The state after executing a block
+
+        Returns
+        -------
+        StateDiff
+            A StateDiff object containing:
+            - 'account_diffs': Dict[Address, AccountDiff] - Accounts that were added, modified or deleted with their storage changes
+        """
+        return compute_state_diff(pre_state, post_state)
+
+
+def compute_state_diff(pre_state: State, post_state: State) -> StateDiff:
```

Comment:
remove all logs related to fork version

Reply:
do you mean the log elated to EIP6780? 

```python
logger.error(
            f"Warning: Since EIP-6780, we should not see any accounts marked for deletion at the level of the block - found {len(deletions)}"
        )
``` 
?

Reply:
yes, this is a Trie lib, focused on trie management and diff. These diff may or may not be consistent with EIPs, but it's another topic